### DISCUSSION TITLE
How to start multi-client connecting to the same server?

### DISCUSSION THREAD

**Discussion by Avaritia1** [eclipse-glsp/glsp#1034] [Q&A]
Hi!

I found this discussion [Whether THEA server can handle multiple people request or dedicated server for each user? #433](https://github.com/eclipse-glsp/glsp/discussions/433). I thought this is kind of similar to my on-going project. I am working on this [EMF.cloud-based server](https://github.com/eclipse-emfcloud/modelserver-glsp-integration/tree/main/project-templates/modelserver-glspjava-emf-theia/server) and this [client](https://github.com/eclipse-glsp/glsp-examples/tree/master/workflow/glsp-client).

And I wonder if there is any solution for the server handling such situation for now?

It seems that if I started a second client listening to the same port, it would not show the graphic editor on the second Theia client.

**Comment by Avaritia1** [comment]
Hi!

A small addition: I found this piece of document in [GLSP Protocol](https://www.eclipse.org/glsp/documentation/protocol/):

> Initialize Request

> The initialize request has to be the first request from the client to the server. Until the server has responded with an InitializeResult no other request or notification can be handled and is expected to throw an error. A client is uniquely identified by an applicationId and has to specify on which protocolVersion it is based on. In addition, custom arguments can be provided in the args map to allow for custom initialization behavior on the server. The request returns an InitializeResult that encapsulates server information and capabilities. The InitializeResult is used inform the client about the action kinds that the server can handle for a specific diagramType.

So if the server can monitor the second port and receive the `Initialize Request` of the second client, can the collaboration between the two clients be realized? If possible, which files of the [model server project](https://github.com/eclipse-emfcloud/modelserver-glsp-integration/tree/main/project-templates/modelserver-glspjava-emf-theia) or [workflow example project](https://github.com/eclipse-glsp/glsp-examples/tree/master/workflow) need to be customized?

**Reply by Avaritia1** [reply]
Can anyone help me with this question? Any insights or ideas are appreciated

**Reply by ndoschek** [reply]
Hi again @Avaritia1!
We will have a look into this this week and will come back to you asap!
Best regards, Nina

**Reply by ndoschek** [reply]
Hi @Avaritia1,

Unfortunately there is currently no solution or example project dealing with multi user collboration.
Overall, the Model Server and/or GLSP Server are not designed for multi user editing (yet).

However, there are other possibilities as well you could think of.
For example, the possibility to collaborate via [VS Code live share](https://code.visualstudio.com/learn/collaboration/live-share).
I know there are currently some experiments going on to use one GLSP server instance and multiple users can join the session to directly work on the same diagram.
However, I'm not sure how far they are already about going public.

HTH and best regards,
Nina
